### PR TITLE
Correctly handle multi container TaskDefinition updates

### PR DESCRIPTION
**Pull request by walterbm**
# What

Update `UpdateTaskDefinitionImage` flow to search through all the containers in a TaskDefinition for the one that is using the ECR repo url before updating the version.

# Why
Previously Outback was naively grabbing the first container in the TaskDefinition and assuming that was always using the correct docker image url. This worked well for single container services but when a TaskDefinition has multiple containers assuming the order is dangerous and lead to situations where the wrong container had their image sha updated.

# How
Instead of always grabbing the first container (i.e. `ContainerDefinitions[0]`) search through the `ContainerDefinitions` for the one that has an `Image` url that contains the `repo` url found in the outback config.

To get this working for `DeployAll` I had to pass the `repo` through a whole bunch of methods but that seems preferable than trying to use some heuristic to determine the repo

### CODE REVIEW

**Review by jclarkkoala — APPROVED** [review]
Looks good!

File: pkg/outback/outback_test.go [hunk 1]
```diff
@@ -1132,6 +1132,54 @@ func TestOutbackRunTaskError(t *testing.T) {
...
+	cases := []string {"other-container:999", "target-container:123"}
+
+	for i, expectedTaskDefinitionImage := range cases {
+		if *result.ContainerDefinitions[i].Image != expectedTaskDefinitionImage {
+			t.Errorf("expected %v value, got %v", expectedTaskDefinitionImage, *result.ContainerDefinitions[i].Image)
+		}
+	}
+}
+
 // func TestOutbackDeploy(t *testing.T) {
```
Comment (Vpr99) [inline pkg/outback/outback_test.go]: I know it's unrelated, but should we nix this other test?

Comment (walterbm) [reply pkg/outback/outback_test.go]: Good point! I don't really know when those tests were added or why they were removed but better to keep the repo clean 🧹

Comment (walterbm) [reply pkg/outback/outback_test.go]: done!

**Pull request merged**